apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V97__Synthea_Pre_Validation.sql

```diff
@@ -0,0 +1,322 @@
+--    -1000010, -1000020, -100000547, -100001170, -100000623, -100000388,
+--    -100000159, 'T01000010A', '1S00E00AA10');
+--
+-- or using named notation like this:
+--
+-- SELECT synthea_load_pre_validate(
+--    p_beg_bene_id => -10000010050073,
+--    p_beg_bene_id => -10000010050083,
+--    p_clm_id => -10000920860670,
+--    p_beg_clm_grp_id => -1617752175,
+--    p_pde_id_start => -10596891505,
+--    p_carr_clm_cntl_num_start => -694670632,
+--    p_fi_doc_cntl_num_start => 423014560,
+--    p_hicn_start => 'T11100073A',
+--    p_mbi_start => '1S00EU2GA73');
+
+-- Function that takes in data values derived from the Synthea
+-- end_state.properties file, and performs a validation check to
+-- determine if the the pipeline can go ahead with processing
+-- the associated RIF files or not.
+--
+-- Basically it performs range-checking on all the claims tables
+-- and beneficiary-related tables. The function returns an integer
+-- value that denotes an OK to proceed (value zero) or not OK to
+-- proceed (value > 0). It does this mainly by getting counts of
+-- rows for various tables and summing the counts; a value of zero
+-- denotes no potential duplicate rows.
+--
+-- The Synthea manifest holds end-state information like the following:
+-- ====================================================================
+--  <endStateProperties>
+--    <numGenerated>1000</numGenerated>
+--    <generated>Tue Oct 18 15:24:11 EDT 2022</generated>
+--    <clm_grp_id_start>-100001170</clm_grp_id_start>
+--    <pde_id_start>-100000623</pde_id_start>
+--    <carr_clm_cntl_num_start>-100000388</carr_clm_cntl_num_start>
+--    <fi_doc_cntl_num_start>-100000159</fi_doc_cntl_num_start>
+--    <hicn_start>T01000010A</hicn_start>
+--    <bene_id_start>-1000010</bene_id_start>
+--    <clm_id_start>-100000547</clm_id_start>
+--    <mbi_start>1S00E00AA10</mbi_start>
+--  </endStateProperties>
+-- ====================================================================
+-- so to test this function, it could be invoked using positional
+-- notation like this:
+--
+-- SELECT synthea_load_pre_validate(
+--    -1000010, -1000020, -100000547, -100001170, -100000623, -100000388,
+--    -100000159, 'T01000010A', '1S00E00AA10');
+--
+-- or using named notation like this:
+--
+-- SELECT synthea_load_pre_validate(
+--    p_beg_bene_id => -10000010050073,
+--    p_beg_bene_id => -10000010050083,
+--    p_clm_id => -10000920860670,
+--    p_beg_clm_grp_id => -1617752175,
+--    p_pde_id_start => -10596891505,
+--    p_carr_clm_cntl_num_start => -694670632,
+--    p_fi_doc_cntl_num_start => 423014560,
+--    p_hicn_start => 'T11100073A',
+--    p_mbi_start => '1S00EU2GA73');
+--
+-- NOTE: The final SQL is not as performant as a version using only PSQL;
+--       Unfortunately, HSQL constraints precluded aspects of the PSQL query
+--       so the RSLT variable had to be incremented in a sequenctial manner.
+--
+${logic.psql-only}  CREATE OR REPLACE FUNCTION synthea_load_pre_validate (
```

Comment:
can you explain some about why a stored procedure was chosen as opposed to just directly executing the select queries from the pipeline?

Reply:
1. SQL is significantly long....
  - so it'll need to be compiled in the db engine every time it is invoked
  - it'll take up 100+ lines of code in the .java source
  - as a function it'll be compiled once, when it it created in the db
 2.  the function actually allows for a 'fail fast' mode of processing (i.e., select progression of queries...first one that fails we're out). Yes, that same could be done .java source, but then that would mean making 10+ queries from Java, each of which will need to be compiled before executing.
 3. it allows for distribution of responsibilities....this is entirely a database pre-validation so let the db do that.

Reply:
I can try....it'll mean doing everything that the SQL function does but in Java source. I do know from running Logan's .py (prepare_and_run_synthea) that the pure db function vs. separate queries ran about 1x faster.

So if I am going to muck up that java code to just verify those numbers, then I might as well just put it all in Java.

Reply:
Well, I think the answer would be that it saves a few milliseconds or seconds at best which would put this in the category of an unnecessary premature optimization. This is only going to run 4 times a year.

Reply:
And I may have underestimated the amount of Java code needed. So the queries themselves will probably up to 100+ lines in Java; then add in the code for CallableStatement and setting variables times the number of calls we'll need to make, probably adds up to 200-300 lines of code in the CcWRifLoadJob.

Reply:
well there are almost 600 lines between the stored procedure and its supporting test. I really don't think that SPL is any less dense than Java and definitely not to the point of justifying the complexity of splitting code across two systems and languages.

Reply:
moved to Java code.